packages/wonder-blocks-data/components/data.js

```diff
@@ -136,14 +136,23 @@ export default class Data<TOptions, TData: ValidData> extends React.Component<
                     const getEntry = this._getCacheLookupFnFromInterceptor(
                         interceptor,
                     );
+
+                    /**
+                     * Need to share our types with InternalData so Flow
+                     * doesn't need to infer them and find mismatches.
+                     */
+                    class TypedInternalData extends InternalData<
+                        TOptions,
+                        TData,
+                    > {}
```

Comment:
Took me a while to work out how to resolve the errors that were coming up here. I realised it was quite possibly because we weren't explicitly parameterizing `InternalData` - so I tried this and it worked.